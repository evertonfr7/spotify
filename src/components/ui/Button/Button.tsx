import { cn } from '@/utils/cn'
import { ButtonProps } from './Button.types'

export function Button({
  children,
  className,
  disabled,
  onClick,
  'aria-label': ariaLabel,
  'aria-describedby': ariaDescribedBy,
  type = 'button',
}: ButtonProps): JSX.Element {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      className={cn(
        'bg-[#57B660] text-[#181414] rounded-3xl px-8 min-h-[32px] font-default text-[14px]/[18px] font-bold md:min-h-[42px] md:px-10 md:text-base/[20px]',
        className,
        disabled && 'opacity-50 cursor-not-allowed',
        !disabled &&
          'focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-[#57B660]',
      )}
      type={type}
      aria-label={ariaLabel}
      aria-describedby={ariaDescribedBy}
      aria-disabled={disabled}
    >
      {children}
    </button>
  )
}
