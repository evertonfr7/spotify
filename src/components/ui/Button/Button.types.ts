export interface ButtonProps {
  children: React.ReactNode;
  className?: string;
  disabled?: boolean;
  onClick?: () => void;
  'aria-label'?: string;
  'aria-describedby'?: string;
  type?: 'button' | 'submit' | 'reset';
}
