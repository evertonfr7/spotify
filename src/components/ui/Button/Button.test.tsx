import { render, fireEvent, screen } from '@testing-library/react'
import { Button } from './Button'

describe('Button component', () => {
  test('renders the button with correct snapshot', () => {
    const tree = render(<Button>Click Me</Button>)

    expect(tree).toMatchSnapshot()
  })

  test('renders the button with children text', () => {
    render(<Button onClick={() => {}}>Click Me</Button>)

    expect(screen.getByText('Click Me')).toBeInTheDocument()
  })

  test('applies additional class names', () => {
    render(
      <Button className="extra-class" onClick={() => {}}>
        Styled Button
      </Button>,
    )

    const button = screen.getByText('Styled Button')
    expect(button).toHaveClass('extra-class')
  })

  test('disables the button when disabled prop is true', () => {
    render(
      <Button disabled={true} onClick={() => {}}>
        Disabled Button
      </Button>,
    )

    const button = screen.getByText('Disabled Button')
    expect(button).toBeDisabled()
    expect(button).toHaveClass('opacity-50')
    expect(button).toHaveClass('cursor-not-allowed')
  })

  test('calls the onClick handler when clicked', () => {
    const handleClick = jest.fn()
    render(<Button onClick={handleClick}>Clickable Button</Button>)

    const button = screen.getByText('Clickable Button')
    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  test('does not call onClick when button is disabled', () => {
    const handleClick = jest.fn()
    render(
      <Button disabled={true} onClick={handleClick}>
        Non-Clickable Button
      </Button>,
    )

    const button = screen.getByText('Non-Clickable Button')
    fireEvent.click(button)
    expect(handleClick).not.toHaveBeenCalled()
  })
})
