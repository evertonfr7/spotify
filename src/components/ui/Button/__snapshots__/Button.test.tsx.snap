// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders the button with correct snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="bg-[#57B660] text-[#181414] rounded-3xl px-8 h-[32px] font-default text-[14px]/[18px] font-bold md:h-[42px] md:px-10 md:text-base/[20px] focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-[#57B660]"
        type="button"
      >
        Click Me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="bg-[#57B660] text-[#181414] rounded-3xl px-8 h-[32px] font-default text-[14px]/[18px] font-bold md:h-[42px] md:px-10 md:text-base/[20px] focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-[#57B660]"
      type="button"
    >
      Click Me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
